### PR TITLE
[BugFix] Fix be-bugs found by sonar

### PR DESCRIPTION
**Pull request by liuyehcf**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

### Rule ["std::move" and "std::forward" should not be confused](https://rules.sonarsource.com/cpp/RSPEC-5417)

If and only if `T` is type template parameter of a function, then `T&&` can be a forwarding reference. For example, the following two cases are both not forwarding reference.

* `std::vector<T>&&` only matches right reference, because it is not in `T&&` mode
* `T&&` only matches right reference, because `T` can be determined when instantiated A, and deduction is nolong required.

```cpp
template<typename T>
void func(std::vector<T>&& v) {xxx}

template<typename T>
class A {
    void func(T&& t) {}
};
```

### ~Rule [Destructors should not throw exceptions](https://rules.sonarsource.com/cpp/RSPEC-1048)~

~The sonar framework has detected some of the destructors may throw exception. And I have tried to add `noexcept` to see whether it can be fixed, but it truns out not to work.~

~So I add `try{} catch(...)` as recommended to tackle this problem.~

### Rule [Unary minus should not be applied to an unsigned expression](https://rules.sonarsource.com/cpp/RSPEC-876)

Given the identical equation `-x = ~x + 1`, I have replaced all the unary minus operator according to the equation.

### Rule ["reinterpret_cast" should be used carefully](https://rules.sonarsource.com/cpp/RSPEC-5275)

Better to use `static_cast` to cast from `short/bool/int/int64` to `int32/int64/int128`, otherwise may produce undefined behavior.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

### CODE REVIEW

File: be/src/exec/vectorized/aggregator.h [hunk 1]
```diff
@@ -145,8 +145,12 @@ class Aggregator : public pipeline::ContextWithDependency {
     Aggregator(const TPlanNode& tnode);
 
     virtual ~Aggregator() {
-        if (_state != nullptr) {
-            close(_state);
+        try {
```
Comment (imay) [inline be/src/exec/vectorized/aggregator.h:148]: Could you explain why this `try catch` is needed?

Comment (liuyehcf) [reply be/src/exec/vectorized/aggregator.h:148]: Add explanation in the `Problem Summary` section

**Comment by liuyehcf** [comment]
run starrocks_be_unittest

**Review by kangkaisen — APPROVED** [review]

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Review by Seaven — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Review by ZiheLiu — APPROVED** [review]

**Pull request merged**